GPU/Common/VertexShaderGenerator.cpp

```diff
@@ -1161,7 +1172,7 @@ bool GenerateVertexShader(const VShaderID &id, char *buffer, const ShaderLanguag
 			WRITE(p, "  %sv_fogdepth = (viewPos.z + u_fogcoef.x) * u_fogcoef.y;\n", compat.vsOutPrefix);
 	}
 
-	if (vertexRangeCulling) {
+	if (vertexRangeCulling && !IsVRBuild()) {
```

Comment:
Some games will get black screens or otherwise without this - the PSP entirely culls any triangle that has a point outside the 4096x4096 grid it considers, unless depth clamp is enabled which causes triangles to potentially be clipped against depth.

Games rely on this and implementing this has fixed bugs where broken geometry incorrectly covers the scene.  Sometimes games render garbage and the bug was never caught in the game because the PSP culled it anyway.

-[Unknown]

Reply:
The problem is that `vertexRangeCulling` causes in VR a huge floor flickering especially in racing games.

I will take a look into it deeper once I found an example of a game which is broken due to this.

Reply:
Well, I guess it will cull things outside of view which is a problem, just noting the other side of that.  Also, some games intentionally "cull" verts in software by moving them to a place they know this culling will happen.  I think there was some racing game with some funky stuff happening in the bottom right offscreen for this reason.

-[Unknown]